core/io/resource.cpp

```diff
@@ -169,10 +174,34 @@ Error Resource::copy_from(const Ref<Resource> &p_resource) {
 			continue; //do not change path
 		}
 
-		set(E.name, p_resource->get(E.name));
+		Variant value = p_from->get(E.name);
+
+		if (p_keep_local_to_scene_resource) {
+			Ref<Resource> sub_res_of_from = value;
```

Comment:
Please add a nice comment here explaining the purpose of all this logic. Let's try to at least keep new additions to this packed scene tangle documented.